### ISSUE TITLE
sort_by on non variable arrays

### ISSUE THREAD

**Issue by dpacino** [MiniZinc/libminizinc#63]
I have read the documentation and searched the forum. From what I can see in the documentation the build-in function "sort_by" should also work on non-variable arrays.  I made the following code snippet:

``` javascript
set of int: N = {1,2,3,4};
array[N] of int: attribute = [100,300, 20, 30];

array[N] of int: sorted_N = sort_by(index_set(N),attribute);
```

When I run the code I get the following error:

```
MiniZinc: type error: initialisation value for `sorted_N' has invalid type-inst: expected `array[int] of int', actual `array[int] of var int'
```

I was expecting

```
function array [int] of $T: sort_by(array [int] of $T: x, array [int] of int: y)
```

to cover this case. Am I missunderstanding something?

**Comment by guidotack** [comment]
Can you please let me know which version of MiniZinc you're using exactly, which operating system, and with which solver?  I can't reproduce the bug here with 2.0.8.

**Comment by dpacino** [comment]
I am using 2.0.8 on OS X Yosemite 10.10.5. I solved the example using Gecode 4.4.0 or G12-fd. I get the same error.

**Comment by Dekker1** [comment]
This issue is being closed in a cleanup of our GitHub repository. It seems it
has been fully discussed to not be an (compiler) issue, already solved, or no
longer relevant.

If this not be the case, feel free to comment below and we'll reconsider the
issue again.
